src/re_frame/interceptor.cljc

```diff
@@ -61,10 +62,22 @@
 ;; -- Execute Interceptor Chain  ------------------------------------------------------------------
 
 
+(defn- exception->ex-info [e interceptor direction]
+  (ex-info (str "Interceptor Exception: " #?(:clj (.getMessage e) :cljs (ex-message e)))
+           {:direction direction
+            :interceptor (:id interceptor)
+            :exception e}))
+
+
 (defn- invoke-interceptor-fn
   [context interceptor direction]
   (if-let [f (get interceptor direction)]
-    (f context)
+    (if (registrar/get-handler :error :event-handler)
+      (try
+        (f context)
+        (catch #?(:clj Exception :cljs :default) e
+          (throw (exception->ex-info e interceptor direction))))
```

Comment:
I agree that the exception looks better, providing more information, but I thought this way would be more backwards compatible. Doesn't seem like people depends on catching the raw exception in `window.onerror` but I don't know...

Also, we could pass something like a `augment-exceptions?` parameter down from `execute` to avoid checking the handler twice.

But maybe it's better to just try/catch always as you suggest. Please confirm and I can do that change

Reply:
> should we always try/catch here? It seems like it would give better exceptions regardless

This would throw an `ExceptionInfo` instead of an `Error`. While the `ExceptionInfo` contains more information, it's debatable whether it "provides more", or "looks better". Exceptions in Clojure are a rather advanced topic, and a new user, who doesn't know/care about them, would have a hard time understanding the `ExceptionInfo` as it appears in the browser console.

I added a commit where `execute` passes both the original `Error` and the augmented `ExceptionInfo` to the handler. Without a handler, it still throws the simple `Error`. This way, power users can do what they want, and new users get the nice error readout.

Reply:
> pass something like a augment-exceptions? parameter down from execute

https://github.com/day8/re-frame/commit/1cbc3c69b65d84a7b5045c62d9826591a8aa14e2